src/game/common/rts/academystats.cpp

```diff
@@ -13,34 +13,761 @@
+    }
+
+    if (m_incomeFrame2 > 3600 || m_incomeFrame1 == 0) {
+        advice_count++;
+
+        if (give_advice && Get_Client_Random_Value(0, count - 1) < max_advice_count - info->m_adviceCount) {
+            info->m_adviceString[info->m_adviceCount].Concat(g_theGameText->Fetch("ACADEMY:NoIncome"));
+            info->m_adviceCount++;
+        }
+
+        count--;
+    }
+
+    if (g_thePlayerList->Get_Local_Player()->Get_Academy_Stats()->Get_Traps_Used() != 0 && m_minesClearedCount == 0) {
+        advice_count++;
+
+        if (give_advice && Get_Client_Random_Value(0, count - 1) < max_advice_count - info->m_adviceCount) {
+            // BUGFIX Change string here from NoIncome to ClearMines (clearly the intended string)
+            info->m_adviceString[info->m_adviceCount].Concat(g_theGameText->Fetch("ACADEMY:ClearMines"));
+            info->m_adviceCount++;
+        }
+
+        count--;
+    }
+
+    if (g_thePlayerList->Get_Neutral_Player()->Get_Academy_Stats()->Get_Killed_Pilot_Count() != 0
+        && m_unmannedVehiclesStolenCount == 0) {
+        advice_count++;
+
+        if (give_advice && Get_Client_Random_Value(0, count - 1) < max_advice_count - info->m_adviceCount) {
+            info->m_adviceString[info->m_adviceCount].Concat(g_theGameText->Fetch("ACADEMY:UnmannedVehicles"));
+            info->m_adviceCount++;
+        }
+
+        count--;
+    }
+
+    if (m_disguiserCount != 0 && m_disguisedUnitCount == 0) {
+        advice_count++;
+
+        if (give_advice && Get_Client_Random_Value(0, count - 1) < max_advice_count - info->m_adviceCount) {
+            info->m_adviceString[info->m_adviceCount].Concat(g_theGameText->Fetch("ACADEMY:DisguisedUnits"));
+            info->m_adviceCount++;
+        }
+
+        count--;
+    }
+
+    if (m_player->Get_Player_Template() != nullptr) {
+        if (m_player->Get_Player_Template()->Get_Base_Side().Compare_No_Case("China") && m_firestormCount == 0) {
+            advice_count++;
+
+            if (give_advice && Get_Client_Random_Value(0, count - 1) < max_advice_count - info->m_adviceCount) {
+                info->m_adviceString[info->m_adviceCount].Concat(g_theGameText->Fetch("ACADEMY:Firestorm"));
+                info->m_adviceCount++;
+            }
+
+            count--;
+        }
+    }
+
+    if (!give_advice && advice_count > 0) {
+        Evaluate_Tier_3_Advice(info, advice_count);
+    }
+}
+
+bool AcademyStats::Calculate_Academy_Advice(AcademyAdviceInfo *info)
+{
+    int max_advice_count = AcademyAdviceInfo::MAX_ACADEMY_ADVICE;
+
+    if (m_skipCalc) {
+        return false;
+    }
+
+    if (info == nullptr) {
+        captainslog_dbgassert(false, "AcademyStats::Calculate_Academy_Advice() was passed in NULL AcademyAdviceInfo.");
+        return false;
+    } else {
+        info->m_adviceCount = 0;
+
+        for (int i = 0; i < max_advice_count; i++) {
+            Utf16String str;
+            str = "\n\n";
+            info->m_adviceString[i].Format(str);
```

Comment:
This looks odd. It effectively just assigns \n\n to advice string?

Reply:
This is correct.